spec/lib/account-manager.spec.js

```diff
@@ -29,44 +40,158 @@ describe("accountManager", () => {
         });
     });
-    it("creates an account", (done) => {
-        var manager, record;
+    it("configured the storage service correctly", () => {
+        factory();
+        expect(storageServiceFactoryMock).toHaveBeenCalledWith({
+            idHash: "goes here"
+        }, {
+            lifetime: "goes here"
+        }, "anything");
+    });
+    describe(".createAsync()", () => {
+        it("creates an account", () => {
+            var record;
 
-        record = {
-            email: "test@example.com",
-            record: true
-        };
-        manager = factory();
-        manager.createAsync(record).then((result) => {
-            expect(randomMock.idAsync).toHaveBeenCalledWith(10);
-            expect(storageServiceFactoryMock.instance.putAsync).toHaveBeenCalledWith("BBBBBBBBBB", record, {
-                email: record.email
+            record = {
+                email: "test@example.com",
+                record: true
+            };
+
+            return factory().createAsync(record).then((result) => {
+                expect(randomMock.idAsync).toHaveBeenCalledWith(10);
+                expect(storageServiceFactoryMock.instance.putAsync).toHaveBeenCalledWith("BBBBBBBBBB", record, {
+                    email: record.email
+                });
+                expect(result).toBe("BBBBBBBBBB");
             });
-            expect(result).toBe("BBBBBBBBBB");
-        }).then(done, done);
+        });
     });
-    it("creates a password hash structure", (done) => {
-        var manager;
+    describe(".loginAsync()", () => {
+        beforeEach(() => {
+            storageService.getAsync.andReturn(promiseMock.resolve({
+                mfa: {
+                    totp: {
+                        key: "abcdefg"
+                    }
+                },
+                passwordHash: "passwordHash"
+            }));
+            challengeManagerMock.validateAsync.andReturn(promiseMock.resolve());
+        });
+        it("fails if TOTP does not validate", () => {
+            totpMock.verifyCurrent.andReturn(false);
 
-        manager = factory();
-        manager.passwordHashConfigAsync().then((passwordHash) => {
-            expect(passwordHash).toEqual({
-                algorithm: "algorithm",
-                derivedLength: "derivedLength",
-                encoding: "encoding",
-                iterations: "iterations",
-                salt: "BBBBBBBBBBBBBBBBBBBB",
-                type: "type"
+            return factory().loginAsync("id", {
+                challengeHash: "challengeHash",
+                mfa: {
+                    totp: "012345"
+                }
+            }).then(() => {
+                jasmine.fail();
+            }, (err) => {
+                expect(err.toString()).toContain("MFA TOTP did not validate");
+                expect(sessionManagerMock.createAsync).not.toHaveBeenCalled();
+            });
+        });
+        it("grabs the record and confirms TOTP", () => {
```

Comment:
This should be saying something about confirming the password and challenge. I don't see this doing anything with TOTP.